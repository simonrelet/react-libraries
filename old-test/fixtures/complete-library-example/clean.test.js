'use strict'

const fs = require('fs-extra')
const path = require('path')
const generatedFolders = require('../../../lib/generatedFolders')
const { Fixture } = require('../../fixtureUtils')

const TEST_FOLDERS = generatedFolders.concat(['custom-folder'])

describe(`${path.basename(__dirname)} clean`, () => {
  let fixture

  beforeAll(async () => {
    fixture = new Fixture(__dirname)
    await fixture.initialize()

    const foldersPath = TEST_FOLDERS.map((folder) =>
      path.join(fixture.paths.root, folder),
    )

    await Promise.all(foldersPath.map((folderPath) => fs.ensureDir(folderPath)))
  })

  afterAll(async () => {
    await fixture.reset()
  })

  it('should not fail', async () => {
    expect(
      await fixture.runScript('clean', ['custom-folder']),
    ).toMatchSnapshot()
  })

  TEST_FOLDERS.forEach((testFolder) => {
    it(`should remove folder: "${testFolder}"`, async () => {
      expect(await fs.exists(path.join(fixture.paths.root, testFolder))).toBe(
        false,
      )
    })
  })
})
