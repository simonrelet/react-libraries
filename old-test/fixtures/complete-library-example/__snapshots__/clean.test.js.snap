// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complete-library-example clean should not fail 1`] = `
Exit code: 0

--- Output ---
Deleted files: build, build-storybook, coverage, custom-folder.
--------------
`;
